daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -952,12 +952,15 @@ private[lf] object SBuiltin {
           stakeholders = cached.stakeholders,
           key = cached.key,
           version = machine.tmplId2TxVersion(cached.templateId),
-        )
-
-      onLedger.updateCachedContracts(coid, cached)
-      onLedger.ptx = newPtx
-      checkAborted(onLedger.ptx)
-      machine.returnValue = SContractId(coid)
+        ) match {
+        case Left((newPtx, err)) =>
+          onLedger.ptx = newPtx // Seems wrong. But one test in ScriptService requires this.
```

Comment:
Which test ?

Reply:
```
bazel run //compiler/damlc/tests:script-service -- -p 'failing transactions'
```

Reply:
The failure is because of a small change to the output in this error case.

It would be great to not care and make this change!